lib/mixpanel-node.js

```diff
@@ -158,15 +167,111 @@ var create_client = function(token, config) {
+                "properties": {
+                    "time": new Date()  // Number or Date; required for each event
+                }
+            },
+            ...
+        ]
+
+        See import() for further information about the import endpoint.
+
+        Options:
+            max_batch_size: the maximum number of events to be transmitted over
+                            the network simultaneously. useful for capping bandwidth
+                            usage.
+
+        N.B.: the Mixpanel API only accepts 50 events per request, so regardless
+        of max_batch_size, larger lists of events will be chunked further into
+        groups of 50.
+
+        event_list:array                    list of event names and properties
+        options:object                      optional batch configuration
+        callback:function(error_list:array) callback is called when the request is
+                                            finished or an error occurs
+    */
+    metrics.import_batch = function(event_list, options, callback) {
+        var MP_BATCH_SIZE = 50; // Mixpanel API permits 50 events per request
+
+        var data_length = event_list.length,
+            total_requests = Math.ceil(data_length / MP_BATCH_SIZE),
+            max_simultaneous_requests = total_requests,
+            completed_requests = 0,
+            event_group_idx = 0,
+            request_errors = [];
+
+        if (typeof(options) === 'function' || !options) {
+            callback = options;
+            options = {};
+        }
+        if (options.max_batch_size) {
+            max_simultaneous_requests = Math.min(
+                total_requests, Math.max(Math.floor(options.max_batch_size / MP_BATCH_SIZE), 1)
+            );
         }
 
-        properties.time = time;
+        var send_batch = function() {
+            var properties,
+                event_batch = [];
 
-        metrics.track(event, properties, callback);
+            for (var ei = event_group_idx; ei < data_length && ei < event_group_idx + MP_BATCH_SIZE; ei++) {
+                properties = event_list[ei].properties;
+                properties.time = parse_time(properties.time);
+                if (!properties.token) {
+                    properties.token = metrics.token;
+                }
+                if (!properties.mp_lib) {
+                    properties.mp_lib = 'node';
+                }
+                event_batch.push(event_list[ei]);
+            }
+            if (event_batch.length > 0) {
+                metrics.send_request('/import', event_batch, function(e) {
+                    completed_requests++;
+                    if (e) {
+                        request_errors.push(e);
+                    }
+                    if (completed_requests === total_requests) {
+                        if (callback) {
+                            callback(request_errors);
+                        }
+                    } else {
+                        send_batch();
```

Comment:
This (and below) is cool, really straightforward way of doing X requests at a time. I like it